src/nifake/tests/test_session.py

```diff
@@ -535,6 +535,16 @@ def test_get_string_ivi_dance_error(self):
                 assert e.code == test_error_code
                 assert e.description == test_error_desc
 
+    def test_get_array_using_ivi_dance(self):
+        self.patched_library.niFake_GetArrayUsingIVIDance.side_effect = self.side_effects_helper.niFake_GetArrayUsingIVIDance
+        self.side_effects_helper['GetArrayUsingIVIDance']['arrayOut'] = None
+        self.side_effects_helper['GetArrayUsingIVIDance']['return'] = 2
+        self.side_effects_helper['GetArrayUsingIVIDance']['arrayOut'] = [1.1, 2.2]
+        self.side_effects_helper['GetArrayUsingIVIDance']['return'] = 0
```

Comment:
You should only be setting 'arrayOut' once with the actual values and none of the others